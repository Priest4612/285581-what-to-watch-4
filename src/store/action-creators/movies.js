import ActionType from '../action-type';

const moviesLoaded = (movies) => ({
  type: ActionType.Movies.FETCH_MOVIES_SUCCESS,
  payload: movies,
});

const setActiveMovieId = (id) => ({
  type: ActionType.Movies.CHANGE_ACTIVE_MOVIE,
  payload: id,
});

export default {
  moviesLoaded,
  setActiveMovieId,
};
