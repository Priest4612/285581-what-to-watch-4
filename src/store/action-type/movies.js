export default {
  FETCH_MOVIES_FAILURE: `FETCH_MOVIES_FAILURE`,
  FETCH_MOVIES_SUCCESS: `FETCH_MOVIES_SUCCESS`,
  FETCH_MOVIES_REQUEST: `FETCH_MOVIES_REQUEST`,
  CHANGE_ACTIVE_MOVIE: `CHANGE_ACTIVE_MOVIE`,
};
