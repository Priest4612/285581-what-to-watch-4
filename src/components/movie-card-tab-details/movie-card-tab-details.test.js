import React from 'react';
import renderer from 'react-test-renderer';

import movies from '../../mocks/test-films-mock.json';

import MovieCardTabDetail from './movie-card-tab-details.jsx';


it(`Render MovieCardTabDetail`, () => {
  const {director, starring, runTime, genre, released} = movies[1];

  const tree = renderer
    .create(<MovieCardTabDetail
      runTime={runTime}
      genre={genre}
      director={director}
      starring={starring}
      released={released}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
