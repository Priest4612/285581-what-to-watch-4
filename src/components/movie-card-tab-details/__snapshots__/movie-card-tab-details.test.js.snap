// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieCardTabDetail 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        Я нёс цветы, дед свою поломанную ветку, мы шли молча.
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        Дэвид Линч
         
        <br />
        Альфред Хичкок
         
        <br />
        Сидни Люмет
         
        <br />
        Квентин Тарантино
         
        <br />
        Майкл Манн
         
        <br />
        Кристофер Нолан
         
        <br />
        Леонид Гайдай
         
        <br />
        Билли Уайлдер
         
        <br />
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        0h 20m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Мишель Родригес
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        1962
      </span>
    </p>
  </div>
</div>
`;
